src/pages/todoList/todoList.js

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from "react";
+import * as T from "./todoList.styles";
+
+export default function TodoListPage() {
+  const [todo, setTodo] = useState("");
+  const [todoList, setTodoList] = useState(() => {
+    // 로컬스토리지에서 할일 불러오기
+    const savedTodo = localStorage.getItem("todoItem");
+    return savedTodo ? JSON.parse(savedTodo) : [];
+  });
+  // 애니메이션 추가,삭제를 위한 인덱스
+  const [addedItemId, setAddedItemId] = useState(null);
+  const [deleteIndexAnimation, setDeleteIndexAnimation] = useState(null);
+
+  // 오늘 날짜
+  const today = new Date().toISOString().slice(0, 10);
+
+  // todoList 변경 시 로컬스토리지 업데이트
+  useEffect(() => {
+    localStorage.setItem("todoItem", JSON.stringify(todoList));
+  }, [todoList]);
+
+  const onChangeTodo = (event) => {
+    setTodo(event.currentTarget.value);
+  };
+
+  const onClickSubmit = (event) => {
+    event.preventDefault();
+    if (!todo.trim()) return;
+
+    const newTodo = { id: Date.now(), text: todo, checked: false };
+    setAddedItemId(newTodo.id);
+    setTodoList([newTodo, ...todoList]);
+
+    // todo 상태 초기화
+    setTodo("");
+  };
+
+  const onClickDelete = (event) => {
+    const indexToDelete = parseInt(event.currentTarget.value, 10);
+
+    setDeleteIndexAnimation(indexToDelete); // 삭제 애니메이션 시작
+    setTimeout(() => {
+      setTodoList(todoList.filter((_, index) => index !== indexToDelete));
+      setDeleteIndexAnimation(null); // 애니메이션 후 상태 업데이트
+    }, 300); // setTimeout의 시간은 애니메이션의 지속 시간과 일치해야 합니다.
+  };
+
+  const onClickToggle = (currentIndex) => {
```

Comment:
토글 함수에서 todolist랑 setTodolist에 직접 접근하는 것도 좋지만,
props로 받아서 수정하는 것이 나중에 협업을 하거나 코드 재사용에 있어서 더 좋을 것 같아요!! 
```suggestion
const onClickToggle = (todoList, setTodoList, currentIndex) => {
```